core/dive.c

```diff
@@ -3020,30 +3020,38 @@ void taglist_cleanup(struct tag_entry **tag_list)
 	}
 }
 
-int taglist_get_tagstring(struct tag_entry *tag_list, char *buffer, int len)
+char* taglist_get_tagstring(struct tag_entry *tag_list)
 {
-	int i = 0;
-	struct tag_entry *tmp;
-	tmp = tag_list;
-	memset(buffer, 0, len);
+	char *buffer = NULL;
+	char *last_pos = NULL;
+	size_t buffer_size = 0;
+	struct tag_entry *tmp = tag_list;
+	/* Compute buffer size counting extra ", " separator(s) */
 	while (tmp != NULL) {
-		int newlength = strlen(tmp->tag->name);
-		if (i > 0)
-			newlength += 2;
-		if ((i + newlength) < len) {
-			if (i > 0) {
-				strcpy(buffer + i, ", ");
-				strcpy(buffer + i + 2, tmp->tag->name);
-			} else {
-				strcpy(buffer, tmp->tag->name);
-			}
-		} else {
-			return i;
+		if (buffer_size > 0) {
+			buffer_size += 2;
```

Comment:
this needs a small change. notice if a tag length is returned as `0` the size will still increment by `2`.
to avoid this store the old size and compare it instead of comparing to `0`.